hera_cal/datacontainer.py

```diff
@@ -324,14 +324,22 @@ def __floordiv__(self, D):
             # iterate over D keys
             for i, k in enumerate(D.keys()):
                 if self.__contains__(k):
-                    newD[k] = self.__getitem__(k) // D[k]
+                    if not (np.any(np.iscomplex(self.__getitem__(k))) or np.any(np.iscomplex(D[k]))):
```

Comment:
Instead of checking if every element is not complex, I wonder if it would be better to use `np.iscomplexobj()` which just checks the datatype.

Reply:
It looks like `iscomplexobj` doesn't work on the dictionary level. Here is a test in numpy 1.22.

```
>>> x = dict({0:np.ones(10, dtype=complex)})
>>> x
{0: array([1.+0.j, 1.+0.j, 1.+0.j, 1.+0.j, 1.+0.j, 1.+0.j, 1.+0.j, 1.+0.j,
       1.+0.j, 1.+0.j])}
>>> np.iscomplexobj(x)
False
```
I've changed the `np.any(np.iscomplex())` bits to `np.iscomplexobj()` at the numpy array level but we still iterate through the dictionary keys.